src/path.c

```diff
@@ -0,0 +1,90 @@
+#include "path.h"
+
+static bool straight = true;
+static char *source;
+static enum movement *destination;
+
+/**
+ * @brief Translate the source path when coming from a straight movement.
+ *
+ * @return Number of movements translated from source.
+ */
+static int from_straight(void)
+{
+	if (!strncmp(source, "F", 1)) {
+		*destination = MOVE_FRONT;
+		return 1;
+	}
+	if (!strncmp(source, "BF", 2)) {
+		*destination = MOVE_START;
+		return 1;
+	}
+	if (!strncmp(source, "LF", 2)) {
+		*destination = MOVE_LEFT_90;
+		return 1;
+	}
+	if (!strncmp(source, "RF", 2)) {
+		*destination = MOVE_RIGHT_90;
+		return 1;
+	}
+	if (!strncmp(source, "LLF", 3)) {
+		*destination = MOVE_LEFT_180;
+		return 2;
+	}
+	if (!strncmp(source, "RRF", 3)) {
+		*destination = MOVE_RIGHT_180;
+		return 2;
+	}
+	straight = false;
+	return 0;
+}
+
+/**
+ * @brief Translate the source path when coming from a diagonal movement.
+ *
+ * @return Number of movements translated from source.
```

Comment:
I would add "pointer iteration" somehow on comments, and maybe include it on the static function